src/main/java/com/arpnetworking/logback/StenoEncoder.java

```diff
@@ -590,6 +606,52 @@ public boolean isSafe() {
     }
 
     /**
+     * Inject bean identifier attributes. This controls whether the the instance identifier and class name are
+     * always injected into each serialized data or context value. The values are always injected for non-loggable
+     * types and this setting allows injection for all types. By default this is false.
+     *
+     * @since 1.9.0
+     *
+     * @param value Whether to inject the bean identifier attributes for all beans irregardless of loggability.
+     */
+    public void setInjectBeanIdentifier(final boolean value) {
+        _injectBeanIdentifier = value;
+    }
+
+    /**
+     * Whether bean identifier attributes are injected. By default this is false.
+     *
+     * @since 1.9.0
+     *
+     * @return True if and only if bean identifier attributes are injected.
+     */
+    public boolean isInjectBeanIdentifier() {
+        return _injectBeanIdentifier;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    protected String encodeAsString(final ILoggingEvent event, final EncodingException ee) {
+        final StringBuilder encoder = new StringBuilder()
+                .append("{\"time\":\"")
+                .append(ISO_DATE_TIME_FORMATTER.format(Instant.ofEpochMilli(event.getTimeStamp())))
+                .append("\",\"name\":\"EncodingException\",\"level\":\"warn\",\"data\":{\"originalMessage\":");
+        safeEncodeValue(encoder, event.getMessage());
+        encoder.append("},\"exception\":{\"type\":\"")
+                .append(ee.getClass().getName())
+                .append("\",\"message\":\"")
+                .append(ee.getMessage().replaceAll("\"", "'"))
```

Comment:
Yes. Good catch. I'll actually delegate this to encodeValue like the other variable data. That should wrap it in Jackson's TextNode causing all encoding to take place. I've also added a test for encoding that newline works as expected.